treeherder/client/thclient/perfherder.py

```diff
@@ -0,0 +1,165 @@
+                property_names.add(property_name)
+        return property_names
+
+    def get_property_values(self, property_name):
+        '''
+        Returns all property values for a particular property name in this collection
+        '''
+        property_values = set()
+        for signature_value in self.signatures.values():
+            if signature_value.get(property_name):
+                property_values.add(signature_value[property_name])
+        return property_values
+
+
+class PerformanceSeries(list):
+    '''
+    Represents a series of performance observations
+
+    You can access the individual elements of the series by using the []
+    syntax. For example, to get the result set ids and geometric means
+    in a summary series:
+
+    ::
+        pc = PerfherderClient()
+        series = pc.get_performance_series('mozilla-central', '9cfc271dab9b7fc2c1229736fecfbbc6e7c5fac9'
+        (result_set_ids, geomeans) = (series['result_set_id'], series['geomean'])
+    '''
+
+    def __getitem__(self, key):
+        return map(lambda el: el[key], self)
+
+
+class PerfherderClient(TreeherderClient):
+
+    PERFORMANCE_SERIES_SUMMARY_ENDPOINT = 'performance-data/0/get_performance_series_summary/?interval={}'
+    SIGNATURE_PROPERTIES_ENDPOINT = 'performance-data/0/get_signature_properties/?signatures={}'
+    PERFORMANCE_DATA_ENDPOINT = 'performance-data/0/get_performance_data/?interval_seconds={}'
+
+    def get_performance_signatures(self, project,
+                                   time_interval=PerformanceTimeInterval.WEEK,
+                                   timeout=None):
+        '''
+        Gets a set of performance signatures associated with a project and time range
+        '''
+        return PerformanceSignatureCollection(self._get_json(
+            project, self.PERFORMANCE_SERIES_SUMMARY_ENDPOINT.format(
+                time_interval), timeout))
+
+    def get_performance_signature_properties(self, project, signature,
+                                             timeout=None):
+        '''
+        Gets the set of properties associated with a specific signature
+        '''
+        endpoint = self.SIGNATURE_PROPERTIES_ENDPOINT.format(signature)
+        property_list = self._get_json(project, endpoint, timeout)
+        if len(property_list) != 1:
+            raise TreeherderClientError(
+                "Expected 1 result for call to '{1}', got '{2}'".format(
+                    endpoint, len(property_list)))
+        return property_list[0]
+
+    def get_performance_series_list(self, project, signature_list,
+                                    time_interval=PerformanceTimeInterval.WEEK,
+                                    timeout=None):
+        '''
+        Gets a list of series objects associated with a set of signatures
+        '''
+        endpoint = self.PERFORMANCE_DATA_ENDPOINT.format(time_interval)
+        for signature in signature_list:
+            endpoint += '&signatures=%s' % signature
```

Comment:
I don't remember if I mentioned it already, but [requests has a `params` argument](http://docs.python-requests.org/en/latest/user/quickstart/#passing-parameters-in-urls) you can use to build the querystring. And it also supports list of parameters :smiley_cat: